subiquity/server/controllers/ad.py

```diff
@@ -33,7 +43,124 @@ async def GET(self) -> Optional[ADConnectionInfo]:
         """Returns the currently configured AD settings"""
         return self.model.conn_info
 
-    async def POST(self, data: ADConnectionInfo) -> List[ADValidationResult]:
+    async def POST(self, data: ADConnectionInfo) -> None:
+        """ Configures this controller with the supplied info.
+            Clients are required to validate the info before POST'ing """
         self.model.conn_info = data
         await self.configured()
-        return [ADValidationResult.OK]
+
+    async def check_admin_name_GET(self, admin_name: str) \
+            -> List[AdAdminNameValidation]:
+        result = AdValidators.admin_user_name(admin_name)
+        return list(result)
+
+    async def check_domain_name_GET(self, domain_name: str) \
+            -> List[AdDomainNameValidation]:
+        result = AdValidators.domain_name(domain_name)
+        return list(result)
+
+    async def check_password_GET(self, password: str) -> AdPasswordValidation:
+        return AdValidators.password(password)
+
+
+# Helper out-of-class functions grouped.
+class AdValidators:
+    """ Groups functions that validates the AD info supplied by users. """
+    @staticmethod
+    def admin_user_name(name) -> Set[AdAdminNameValidation]:
+        """ Validates the supplied admin name against known patterns.
+            Returns a set of the errors detected. """
+        result = set()
+
+        if len(name) == 0:
+            log.debug("admin name is empty")
+            return {AdAdminNameValidation.EMPTY}
+
+        # Ubiquity checks the admin name in two steps:
+        # 1. validate the first char against '[a-zA-Z]'
+        # 2. check the entire string against r'^[-a-zA-Z0-9_]+$'
+        if not re.match('[a-zA-Z]', name[0]):
```

Comment:
`re.match` is strange and acts as if the pattern is prefixed with "^". Overall, I prefer `re.search` and `re.fullmatch` for this reason but in this specific scenario, `re.match` makes sense. The [0] part is not necessary though:

```suggestion
        if not re.match('[a-zA-Z]', name):
```